src/core/model/StrokeStyle.cpp

```diff
@@ -1,91 +1,83 @@
 #include "StrokeStyle.h"
 
 #include <cstring>  // for memcmp, strcmp, strncmp
+#include <iomanip>  // for setprecision
+#include <iterator> // for ostream_iterator
+#include <map>      // for map
+#include <sstream>  // for istringstream
 #include <vector>   // for vector
 
-#include <glib.h>  // for g_ascii_strtod, g_free, g_strdup_printf
-
 #include "model/LineStyle.h"  // for LineStyle
 
 StrokeStyle::StrokeStyle() = default;
 
 StrokeStyle::~StrokeStyle() = default;
 
-const double dashLinePattern[] = {6, 3};
-const double dashDotLinePattern[] = {6, 3, 0.5, 3};
-const double dotLinePattern[] = {0.5, 3};
-
-#define PARSE_STYLE(name, def)                                \
-    if (strcmp(style, name) == 0) {                           \
-        LineStyle style;                                      \
-        style.setDashes(def, sizeof(def) / sizeof((def)[0])); \
-        return style;                                         \
-    }
+namespace {
 
-auto StrokeStyle::parseStyle(const char* style) -> LineStyle {
-    PARSE_STYLE("dash", dashLinePattern);
-    PARSE_STYLE("dashdot", dashDotLinePattern);
-    PARSE_STYLE("dot", dotLinePattern);
+constexpr auto CUST_KEY = "cust: ";
 
+const std::map<std::string, std::vector<double>> predefinedPatterns = {
+    { "dash", {6, 3} }, 
+    { "dashdot", {6, 3, 0.5, 3} },
+    { "dot", {0.5, 3}}
+};
 
-    if (strncmp("cust: ", style, 6) != 0) {
-        return LineStyle();
-    }
+auto formatStyle(const double* dashes, int count) -> std::string {
```

Comment:
This function is not very optimal. I think you could go a little bit deeper into the refactoring and change LineStyle to replace the pair `dashes, count` by an `std::vector<double>`.
The way the LineStyle is used down the road is by passing dashes and count to cairo. This can be done by accessing the std::vector's data directly.

Reply:
Yes,  I am working on LineStyle also. I will add it to this.